### PR TITLE
CLN: json_table_schema remove unused arg, annotate

### PR DESCRIPTION
**Pull request by jbrockmendel**
cc @WillAyd

### CODE REVIEW

**Review by TomAugspurger — APPROVED** [review]

File: pandas/io/json/_table_schema.py [hunk 1]
```diff
@@ -26,17 +27,17 @@
 loads = json.loads
 
 
-def as_json_table_type(x):
+def as_json_table_type(x: DtypeObj) -> str:
```
Comment (WillAyd) [inline pandas/io/json/_table_schema.py:30]: I think this can just be `Dtype`, no?

Comment (jbrockmendel) [reply pandas/io/json/_table_schema.py:30]: DtypeObj is more specific

Comment (WillAyd) [reply pandas/io/json/_table_schema.py:30]: I think Dtype is more correct though; these both seem to work:

```python
>>> as_json_table_type(int)
'integer'
>>> as_json_table_type("int")
'integer'
```

Comment (jbrockmendel) [reply pandas/io/json/_table_schema.py:30]: They do in the status quo, but we never actually pass strings. im working on a branch that addresses #33368 implementing just-the-dtype versions of the is_foo_dtype functions, looking for places where we can rule out anything-but-dtype-objects, stumbled on this

Comment (WillAyd) [reply pandas/io/json/_table_schema.py:30]: Ah OK I see. Makes sense in the grand scheme of things - thanks for clarifying

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @jbrockmendel

**Comment by jreback** [comment]
lgtm.

**Pull request merged**